File: src/js/src/scopes/TeletypeScope.js
function TeletypeScope() {
	this.type = ScopeTypeCache.teletype;
	this.level = ScopeLevel.inline;
	this.canNestText = true;
	
	this.open = function(analyzer) {
		analyzer.openTeletypeText();
	};
	this.close = function(analyzer) {
		analyzer.closeTeletypeText();
	};
}

extend(Scope, TeletypeScope);

File: src/dotnet/Src/ButterflyNet.Parser/Scopes/TeletypeScope.cs
namespace ButterflyNet.Parser.Scopes {
	public class TeletypeScope : InlineScope {
		public override void Open(ButterflyAnalyzer analyzer) {
			analyzer.OpenTeletypeText();
		}

		public override void Close(ButterflyAnalyzer analyzer) {
			analyzer.CloseTeletypeText();
		}
	}
}